go-controller/pkg/ovn/obj_retry.go

```diff
@@ -728,6 +800,18 @@ func (oc *Controller) deleteResource(objectsToRetry *retryObjs, obj, cachedObj i
 			extraParameters.portGroupEgressDenyName,
 			obj)
 
+	case factory.EgressFirewallType:
+		var err error
+		egressFirewall, ok := obj.(*egressfirewall.EgressFirewall)
```

Comment:
get rid of the ok check and just directly assert the type without checking for an error. This should never fail and if it does we should panic as its a glaring bug in our code.